src/Microsoft.Identity.Web.Certificate/DefaultCredentialsLoader.cs

```diff
@@ -69,7 +69,25 @@ public async Task LoadCredentialsIfNeededAsync(CredentialDescription credentialD
                     if (credentialDescription.CachedValue == null)
                     {
                         if (CredentialSourceLoaders.TryGetValue(credentialDescription.SourceType, out ICredentialSourceLoader? loader))
-                            await loader.LoadIfNeededAsync(credentialDescription, parameters);
+                        {
+                            try
+                            {
+                                await loader.LoadIfNeededAsync(credentialDescription, parameters);
+                            }
+                            catch (System.Exception ex)
+                            {
+                                _logger?.LogInformation(ex, $"Failed to load credential {credentialDescription.Id} from source {credentialDescription.SourceType}.");
+
+                                // Certificates are re-attempted, but assertions are not (behavior in Id.Web 3.1)
+                                // It makes sense to skip assertions (e.g. federated identity) because many developers use a "FIC (federated identity credential)" + "certificate fallback" setup
+                                // and we don't want to keep checking the FIC.
+                                if (credentialDescription.CredentialType == CredentialType.SignedAssertion)
+                                {
+                                    _logger.LogDebug("Setting Skip to true");                                    
+                                    credentialDescription.Skip = true;
```

Comment:
I have kept the logic as is, i.e. only the "signed assertion" have this flag set to true. 

However, I think it makes sense to set it always. If a credential fails to load, is there any value in re-trying it? 

Note: this is the root cause of the issue #3000 - this flag wasn't set